libraries/AP_HAL_ChibiOS/hwdef/CubeRedPrimary/hwdef.dat

```diff
@@ -0,0 +1,260 @@
+
+# Nnow some defines for logging and terrain data files.
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# Enable FAT filesystem support (needs a microSD defined via SDMMC).
+define HAL_OS_FATFS_IO 1
+
+# Setup the IMU heater
+define HAL_HAVE_IMU_HEATER 1
+define HAL_IMU_TEMP_DEFAULT 45
+define HAL_IMUHEAT_P_DEFAULT 50
+define HAL_IMUHEAT_I_DEFAULT 0.07
+define HAL_IMU_TEMP_MARGIN_LOW_DEFAULT 5
+
+
+# supports upto 8MBits/s
+CANFD_SUPPORTED 8
+
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+STORAGE_FLASH_PAGE 14
+define HAL_STORAGE_SIZE 32768
+
+# ADC setup
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PF11 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+PB1 AUX_POWER ADC2 SCALE(1)
+PF12 AUX_ADC2 ADC1 SCALE(1)
+
+PF13 VDD_5V_SENS ADC2 SCALE(2)
+
+# And the analog input for airspeed (rarely used these days).
+PF3 PRESSURE_SENS ADC3 SCALE(2)
+PC3 FMU_SERVORAIL_VCC_SENS ADC3 SCALE(3)
+
+# CAN config
+PE10 GPIOCAN1_TERM OUTPUT HIGH
+PG4 GPIOCAN2_TERM OUTPUT HIGH
+
+PB8    CAN1_RX    CAN1
+PB9    CAN1_TX    CAN1
+PB5    CAN2_RX    CAN2
+PB6    CAN2_TX    CAN2
+
+
+# SPI1
+PE9    BARO_0_CS    CS
+PE13   ICM42688_0_CS    CS
+PB3    SPI1_SCK    SPI1
+PB4    SPI1_MISO    SPI1
+PD7    SPI1_MOSI    SPI1
+
+# SPI2
+PD14   RM3100_CS    CS
+PD15   ICM42688_1_CS    CS
+PA9    SPI2_SCK    SPI2
+PB14    SPI2_MISO    SPI2
+PB15    SPI2_MOSI    SPI2
+
+# SPI4
+PF5    ICM20649_CS   CS
+PC2    BARO_1_CS   CS
+PE12    SPI4_SCK    SPI4
+PE5    SPI4_MISO    SPI4
+PE6    SPI4_MOSI    SPI4
+
+# Sensors
+SPIDEV icm42688_0       SPI1 DEVID1  ICM42688_0_CS      MODE3  2*MHZ  8*MHZ
+SPIDEV ms5611_0         SPI1 DEVID2  BARO_0_CS          MODE3 20*MHZ 20*MHZ
+
+SPIDEV icm42688_1       SPI2 DEVID1  ICM42688_1_CS      MODE3  2*MHZ  8*MHZ
+SPIDEV rm3100           SPI2 DEVID2  RM3100_CS          MODE3  2*MHZ  8*MHZ
+
+SPIDEV icm20649         SPI4 DEVID1  ICM20649_CS        MODE3  4*MHZ  8*MHZ
+SPIDEV ms5611_1         SPI4 DEVID2  BARO_1_CS          MODE3 20*MHZ 20*MHZ
+
+IMU Invensensev3 SPI:icm42688_1 ROTATION_PITCH_180_YAW_270
+IMU Invensensev3 SPI:icm42688_0 ROTATION_PITCH_180_YAW_270
+IMU Invensensev2 SPI:icm20649 ROTATION_NONE
+
+BARO MS56XX SPI:ms5611_0
+BARO MS56XX SPI:ms5611_1
+
+COMPASS RM3100 SPI:rm3100 false ROTATION_NONE 
+
+
+# MCO output
+PA8    RCC_MCO_1    INPUT
+
+# I2C
+PF0    I2C2_SDA    I2C2
+PF1    I2C2_SCL    I2C2
+PF14    I2C4_SCL    I2C4
+PF15    I2C4_SDA    I2C4
+I2C_ORDER I2C2 I2C4
+
+# Ethernet
+# PA1    ETH_RMII_REF_CLK    ETH1
+# PA2    ETH_MDIO    ETH1
+# PA7    ETH_RMII_CRS_DV    ETH1
+# PB11    ETH_RMII_TX_EN    ETH1
+# PB12    ETH_RMII_TXD0    ETH1
+# PB13    ETH_RMII_TXD1    ETH1
+# PC1    ETH_MDC    ETH1
+# PC4    ETH_RMII_RXD0    ETH1
+# PC5    ETH_RMII_RXD1    ETH1
+# define BOARD_PHY_ID  MII_LAN8720_ID
+# define BOARD_PHY_RMII
+
+# RCIN
+PB7  TIM4_CH2 TIM4 RCININT  PULLDOWN
+
+# QUADSPI
+PD11    QUADSPI_BK1_IO0    QUADSPI1
```

Comment:
Don't think you need these - no QSPI device in the regular firmware

Reply:
I think you do, the pin config needs to remain. If I don't set it, it will revert to input floating in regular firmware, and comms with QUADSPI will die.

Reply:
We specifically don't reset QSPI in the main firmware otherwise memory mapping stops working, but we must reset it in the bootloader. Somehow we need a mechanism to keep QSPI live in the main firmware without re initializing. The SPRacing board does not have this.